dll/win32/browseui/addressband.cpp

```diff
@@ -259,8 +259,63 @@ HRESULT STDMETHODCALLTYPE CAddressBand::HasFocusIO()
     return S_FALSE;
 }
 
+static WCHAR GetAccessKeyFromText(WCHAR chAccess, LPCWSTR pszText)
+{
+    for (const WCHAR *pch = pszText; *pch != UNICODE_NULL; ++pch)
+    {
+        if (*pch == L'&' && pch[1] == L'&')
+        {
+            ++pch;
+            continue;
+        }
+        if (*pch == L'&')
+        {
+            ++pch;
+            chAccess = *pch;
+            break;
+        }
+    }
+
+    ::CharUpperBuffW(&chAccess, 1);
+    return chAccess;
+}
+
+static WCHAR GetAddressBarAccessKey(WCHAR chAccess)
+{
+    static WCHAR s_chCache = 0;
+    if (s_chCache)
+        return s_chCache;
+
+    WCHAR szText[80];
+    if (!LoadStringW(_AtlBaseModule.GetResourceInstance(), IDS_ADDRESSBANDLABEL,
```

Comment:
Observation: If you minimally modify the GetAccessKeyFromText helper to take a counted string (instead of a NULL-terminated one), you would be able to use LoadStringW with the syntax where it can return directly a pointer to the string and its length in characters, and use that.
This would allow not copying the string into a temporary buffer that has an arbitrary length.